test/src/utils/testRequireModule.js

```diff
@@ -0,0 +1,20 @@
+const {strict: assert} = require('node:assert');
+const path = require('node:path');
+
+const requireModule = require('../../../dist/utils/requireModule.js');
```

Comment:
shouldn't it use the require module from js not dist folder?

Reply:
We need to require the file from `dist` folder only because here we are testing if the `requireModule` is transpiled correctly by `tsc` or not.

Reply:
And in all other tests also, we require the files from `dist` folder only because `dist` is what we ship to the users, so we test if the files present in the `dist` folder are behaving correctly or not.

Reply:
This is the reason why I left typescript and babel too. Only the distributed code matters, so recommended to test it.
I deliver `d.ts` files only for ts and IDE support.

During the last 8 years I used typescript and my experience was that it helped in some areas and caused troubles as well. At the end of the day the sum is 0.

Reply:
Yeah, forgot about that.

@ert78gb We should use the `common` module here instead of using the relative path to `dist`, like below:
```js
const common = require('../../common.js');
const requireModule = common.require('utils/requireModule.js');
```

Reply:
Done.